### ISSUE TITLE
Can't seem to configure :clojure-lsp/unused-public-var at the namespace level

### ISSUE THREAD

**Issue by ikappaki** [clojure-lsp/clojure-lsp#1453]
**Describe the bug**
Can't turn off `:clojure-lsp/unused-public-var` level in namespace metadata under the `:clj-kondo/config` key

**To Reproduce**

1. Create a `src/exp.clj` file with an unused var while turning the `unused-public-var` linter off.
```
```

3. Run clojure-lsp clean-ns --flags... 

**Expected behavior**

<details>
 <summary><b>clojure-lsp output</b></summary>
<pre>
ADD HERE
</pre>
</details>

**User details (please complete the following information):**
 - OS: [e.g. ArchLinux, MacOS, Windows 10]
 - Version: (post the result of `clojure-lsp --version`)

**Additional context**

**Comment by ericdallo** [comment]
This happens because this linter is a clj-kondo custom linter used in `:custom-lint-fn`, I think clj-kondo should support that but not sure how hard would be, WDYT @borkdude ? (I can't recall or find a issue related to that if already exists)

**Comment by borkdude** [comment]
I think it might be better to do this as `:config-in-ns` and then the lsp linter itself can look into the config?

**Comment by ericdallo** [comment]
I was trying to avoid having more custom lint logic on lsp side, since other clj-kondo users can use custom-lint-fn, if they apply a `reg-finding!` they would face the same issue not being able to ns meta config or need to manually read/handle :config-in-ns, right?

**Comment by borkdude** [comment]
Makes sense, but the `reg-finding` is called without any reference to the namespace, so how would clj-kondo know how to suppress that lint warning?

**Comment by ericdallo** [comment]
Good point, maybe we could add a extra arg to that reg-finding?

**Comment by borkdude** [comment]
I don't see the complete picture of this yet, but I'm open to an experimental PR to explore that idea

**Comment by ikappaki** [comment]
Hi @ericdallo,

is there anything I can help you out with this to get it moving?

thanks

**Comment by ericdallo** [comment]
@ikappaki my idea is: currently clj-kondo exposes a `:custom-lint-fn` arg that clojure-lsp passes during every lint, as arg of that function, clj-kondo passes `reg-findin!`, which currently clojure-lsp calls it [here](https://github.com/clojure-lsp/clojure-lsp/blob/master/lib/src/clojure_lsp/feature/diagnostics.clj#L225), I was thinking if clj-kondo could receive on that map arg the `:ns` and internally check the ns config as it does for other linters

**Comment by mainej** [comment]
@borkdude is it correct that clj-kondo collects ns metadata config while a file is being read, uses it while processing that file, and then discards it? 

If so, your plan sounds hard @ericdallo. Consider the following two files:

```clojure
;; using.clj
(ns using
  (:require [defining]))

defining/a-var

;; defining.clj
(ns defining
  {:clj-kondo/config '{:linters {:clojure-lsp/unused-public-var {:level :off}}}})

(def a-var 1)
```

If the user deletes the var-usage from `using.clj`, then the var-definition in `defining.clj` is unused.

However, in this situation we ask clj-kondo to analyze the contents of `using.clj`, but not `defining.clj`. During the analysis run clj-kondo won't ever see the ns metadata config from `defining.clj`, the config which it needs to decide whether to include lint on a var defined in the `defining` ns. 

It should have seen that config in a previous run, whenever `defining.clj` was last analyzed, but that means you need to give clj-kondo a durable cache of namespace -> config (or maybe [filename namespace] -> config) which it can re-use in later runs. It's not impossible, just sounds like quite a lot of work.

I wish I had a better suggestion about how to implement this, but unfortunately I don't. Just pointing out problems 😛

**Comment by borkdude** [comment]
For this situation it's preferred to use `:config-in-ns`
